src/main/java/com/amazon/ion/impl/schubfach/DoubleToDecimal.java

```diff
@@ -264,8 +264,8 @@ private static DoubleToDecimal threadLocalInstance() {
     private String toDecimalString(double v) {
         switch (toDecimal(v)) {
             case NON_SPECIAL: return charsToString();
-            case PLUS_ZERO: return "0.0";
-            case MINUS_ZERO: return "-0.0";
+            case PLUS_ZERO: return "0e0";
+            case MINUS_ZERO: return "-0e0";
             case PLUS_INF: return "Infinity";
             case MINUS_INF: return "-Infinity";
             default: return "NaN";
```

Comment:
These should be `+inf`, `-inf` and `nan` respectively: https://amazon-ion.github.io/ion-docs/docs/float.html#special-values